python/lsst/summit/utils/guiders/seeing.py

```diff
@@ -0,0 +1,517 @@
+            Flattened list of per-detector variances:
+            ``[vx_det1, vy_det1, vx_det2, vy_det2, ...]``. If a detector
+            was excluded or lacks corrected coordinates, entries are
+            ``NaN``.
+        """
+        out: list[float] = []
+        for _, df in self.starList.items():
+            meanFlux = float(np.nanmean(pd.to_numeric(df["flux"], errors="coerce")))
+            hasCorr = "dxfp_corr" in df and "dyfp_corr" in df
+            if meanFlux > self.fluxCut and hasCorr:
+                vx = float(np.nanvar(pd.to_numeric(df["dxfp_corr"], errors="coerce"))) * (100.0**2)
+                vy = float(np.nanvar(pd.to_numeric(df["dyfp_corr"], errors="coerce"))) * (100.0**2)
+                out.extend([vx, vy])
+            else:
+                out.extend([np.nan, np.nan])
+        return out
+
+    @staticmethod
+    def makePairs(kind: str) -> list[list[str]]:
+        """
+        Produce preset detector pairs.
+
+        Parameters
+        ----------
+        kind : `str`
+            One of ``{'adjacent','cross','diagonal'}``.
+
+        Returns
+        -------
+        pairs : `list[list[str]]`
+            List of 2-element lists ``[det1, det2]`` describing pairs.
+
+        Raises
+        ------
+        KeyError
+            If ``kind`` is not a recognized preset.
+        """
+        presets: dict[str, list[list[str]]] = {
+            "adjacent": [
+                ["R00_SG0", "R00_SG1"],
+                ["R04_SG0", "R04_SG1"],
+                ["R40_SG0", "R40_SG1"],
+                ["R44_SG0", "R44_SG1"],
+            ],
+            "cross": [
+                ["R00_SG0", "R40_SG1"],
+                ["R04_SG1", "R44_SG0"],
+                ["R00_SG1", "R04_SG0"],
+                ["R40_SG0", "R44_SG1"],
+            ],
+            "diagonal": [
+                ["R00_SG0", "R44_SG1"],
+                ["R00_SG1", "R44_SG0"],
+                ["R40_SG0", "R04_SG1"],
+                ["R40_SG1", "R04_SG0"],
+            ],
+        }
+        return presets[kind]
+
+    def measureCorrPairs(
+        self, kind: str
+    ) -> tuple[NDArray[np.float64], NDArray[np.float64], NDArray[np.float64]]:
+        """
+        Compute pairwise correlations, covariances, and differential
+        variances between centroid motions of detector pairs.
+
+        The detector pairs are chosen by a preset specified by ``kind``.
+        For each pair of detectors, star centroids are rotated into the
+        local (perpendicular, parallel) frame defined by the vector that
+        connects their mean positions. The method then computes:
+        - Pearson correlation coefficients,
+        - covariance (scaled to centipix²), and
+        - variance of the centroid differences (scaled to centipix²),
+        separately for the perpendicular (x) and parallel (y) components.
+
+        Parameters
+        ----------
+        kind : `str`
+            Preset name passed to :meth:`makePairs`.
```

Comment:
I'm terrible at the automated doc stuff - does that actually work and render OK, and how is it for inline completion in _e.g._ a notebook? If it's not good, just saying `self.makePairs()` would be fine though.